mantle/kola/harness.go

```diff
@@ -1107,8 +1136,73 @@ func collectLogsExternalTest(h *harness.H, t *register.Test, tcluster cluster.Te
 	}
 }
 
+func createTestBuckets(tests []*register.Test) [][]*register.Test {
+
+	// Make an array of maps. Each entry in the array represents a
+	// test bucket. Each corresponding map is the test.Name -> *register.Test
+	// mapping for tests to be executed
+	var bucketInfo []map[string]*register.Test
+
+	// Get a Map of test.Name -> *register.Test
+	testMap := make(map[string]*register.Test)
+	for _, test := range tests {
+		testMap[test.Name] = test
+	}
+
+	// Update all test's conflict lists to be complete
+	// FYI: it is possible that test.Conflicts contain duplicates
+	// this should not affect the creation of the buckets
+	for _, test := range tests {
+		for _, conflict := range test.Conflicts {
+			if _, found := testMap[conflict]; found {
+				testMap[conflict].Conflicts = append(testMap[conflict].Conflicts, test.Name)
+			} else {
+				plog.Debugf("%v specified %v as a conflict but %v was not found. Double-check that it is marked as non-exclusive.",
+					test.Name, conflict, conflict)
+			}
```

Comment:
If a particular conflict specified by another test is not found, previously the logic here would have errored. This change allows us to safely check whether we have seen the test specified as a conflict before.

Reply:
Hmm. the testMap[] contains the information about all available tests, right? Why would we want to not error if someone specifies a conflict that doesn't exist? Wouldn't we want to clean up our code base rather than leave that hanging around?

Maybe there is some valid use case I'm not seeing. I could see an argument for say "I have a test in the RHCOS repo that conflicts with a test from the FCOS repo, but when we run tests against FCOS we will not have loaded the RHCOS tests", but my argument to that is "just declare the conflict in the RHCOS test".

Anything I'm not considering?

Reply:
In the case that someone simply misspells a a test name, I thought a more informative message should be displayed than just a nil pointer error. I didn't want to error out due to the FCOS - RHCOS test conflict case that you mentioned, but now I'm thinking that displaying an explicit error might be the way to go.

Reply:
Agree. Fatal error with more informative error message 👍